llvm/lib/Support/APFloat.cpp

```diff
@@ -1075,6 +1099,11 @@ bool IEEEFloat::isSignificandAllOnesExceptLSB() const {
 }
 
 bool IEEEFloat::isSignificandAllZeros() const {
+  // For the E8M0 format, this is always true since there are no
+  // actual significand bits.
+  if (semantics == &semFloat8E8M0FN)
+    return true;
```

Comment:
Can you avoid adding special case early exits for these? We shouldn't be treating these exotic formats as a common case, and I would hope this would naturally fall out from the code below

Reply:
I would assume it's possible to rework the code to handle 0 part count instead of an explicit check. I'm also wondering if this is exotic enough to be its own subclass of APFloatBase instead of pretending it's IEEEfloat since it's reached a new level of degeneracy

Reply:
Orthogonal to the discussion above but if we moved forward with the early exit, could we implement it using `hasExponentOnly()` to cover any other future cases?

Reply:
I am trying out a few options here, and will update here.

Reply:
We can handle these with precision checks along with the existing conditions.
So, in the latest revision, I have removed all the type-based checks.

Reply:
Matt, Kindly take a look and let me know if the latest revision looks good.

Reply:
I had pinged you in discord last week with some more details and requested another review here.
Thanks for taking another look, Matt!